### DISCUSSION TITLE
Handling checkbox/checklist

### DISCUSSION THREAD

**Discussion by dtrckd** [dillonkearns/elm-markdown#131] [Q&A]
Hello,

I am trying to handle clicks within checkbox (`- [ ] foo bar`) in a markdown that I render. 
I believe that I need the line number corresponding to that item in the raw string in order to proceed.

Do you think it is possible to get that line number with elm-markdown ? 
Or maybe there is a different approach to handle msg for checkbox that has been tried before ?

**Comment by dillonkearns** [comment]
I believe you could create your custom Renderer with the `unorderedList : List (ListItem view) -> view` field rendering to your type of `Html Msg` in the case that the [`ListItem`](https://package.elm-lang.org/packages/dillonkearns/elm-markdown/latest/Markdown-Block#ListItem) is a [`Task`](https://package.elm-lang.org/packages/dillonkearns/elm-markdown/latest/Markdown-Block#Task).

**Reply by dtrckd** [reply]
Yes, this is what I have created. I can handle click on a task, but how are we supposed to know what line has/task has been clicked ??

**Reply by dillonkearns** [reply]
You can do a `List.indexedMap` to get the index within the list.

**Reply by dtrckd** [reply]
But what if they are multiple (group of) checklists in the string ?

**Reply by dillonkearns** [reply]
Good question, that might not be possible to capture with a standard renderer there. I'd be open to a pull request or an issue to explore possible solutions there. I'm not sure if the parser stores that context currently, or what context exactly it should include. The simplest thing I would imagine would be for it to include the nested index level (the parser might not have that information yet). To include the raw item might be harder possibly.

I believe you can accomplish this with the helpers in the [`matheus23/elm-markdown-transforms` package](https://package.elm-lang.org/packages/matheus23/elm-markdown-transforms), like https://package.elm-lang.org/packages/matheus23/elm-markdown-transforms/latest/Markdown-Scaffolded#foldIndexed.

**Reply by dtrckd** [reply]
Thanks for the links Dillon, I am going to explore that.
